source/plugin/injection.rst

```diff
@@ -12,19 +12,23 @@ Dependency injection allows plugins to designate a few API types that will be in
 Temporary List of Injected Types
 ================================
 
-ConfigDir
+ConfigDir (annotation on File or Path)
   Used to inject the plugin's configuration directory:
   ``./mods/`` OR ``./mods/<Plugin#id>/`` depending on ``sharedRoot``
 
-ConfigFile
+ConfigurationLoader<CommentedConfigurationNode>
+  Must be annotated with ``@DefaultConfig``.
+  Used to inject a pre-generated ``ConfigurationLoader`` for the ``File`` of the same annotation.
+
+DefaultConfig (annotation on File, Path or ConfigurationLoader)
   Used to inject the plugin's specific configuration file: ``<Plugin#id>.conf``
 
 EventManager
   Manages the registration of event handlers and the dispatching of events.
 
-File
+File or Path
   This must have an additional annotation specifying specific file!
```

Comment:
> This must have an additional annotation specifying specific file!

You haven't changed this, although this sentence doesn't make sense for me right now ("specifying specific file")